Mathlib/Data/Nat/GCD/Basic.lean

```diff
@@ -88,6 +89,47 @@ theorem gcd_self_add_right (m n : ℕ) : gcd m (m + n) = gcd m n := by
   rw [add_comm, gcd_add_self_right]
 #align nat.gcd_self_add_right Nat.gcd_self_add_right
 
+-- Lemmas where one argument consists of a subtraction of the other
+
+@[simp]
+theorem gcd_sub_self_left (m n : ℕ) (h : m ≤ n) : gcd (n - m) m = gcd n m := by
+  calc
+    gcd (n - m) m = gcd (n - m + m) m  := by rw [← gcd_add_self_left (n - m) m]
+                _ = gcd n m := by rw [Nat.sub_add_cancel h]
+
+@[simp]
+theorem gcd_self_sub_left (m n : ℕ) (h : m ≤ n) : gcd (n - m) m = gcd m n := by
+  rw [gcd_sub_self_left _ _ h, gcd_comm]
```

Comment:
Is `simp` going to ever fire on the second one?  It seems that if it could fire on the second, it would have already fired on the first.

Reply:
That makes sense. Since it is never quite clear what would be the better `simp`, perhaps it would be best to remove  both?! That is what I have done now.